app/models/user.rb

```diff
@@ -6,4 +6,39 @@ class User < ApplicationRecord
 
   has_many :dragons
   has_many :resources
+
+  def can_afford?(dragon_type)
+    resources_list = DragonCost.resources_amount(dragon_type)
+    resources_list.each do |resource|
+      user_amount = resources.find_by(resource_type: resource.resource_type.id)
+      if !resource?(user_amount) || !good_amount?(user_amount, resource)
+        return false
+      end
+    end
+    true
+  end
+
+  def missing_resources_for(dragon_type)
+    missing_resources = 'Lacking resources: '
+    resources_list = DragonCost.resources_amount(dragon_type)
+    resources_list.each do |resource|
+      user_amount = resources.find_by(resource_type: resource.resource_type.id)
+      if !resource?(user_amount)
+        missing_resources += "#{resource.cost} #{resource.resource_type.name} "
+      elsif !good_amount?(user_amount, resource)
+        missing_resources += "#{resource.cost - user_amount.quantity} #{resource.resource_type.name} "
+      end
+    end
+    missing_resources
+  end
+
+  def resource?(resource)
```

Comment:
ta metoda tak naprawdę robi:
```ruby
def resource?(resource)
  !resource.nil?
end
```

więc skoro jest taka prosta to bym ją usunął i skorzystał z tego kodu bezpośrednio w metodzie wyżej:
```ruby
  user_amount = resources.find_by(resource_type: resource.resource_type.id)
  if user_amount.nil?
    missing_resources += "#{resource.cost} #{resource.resource_type.name} "
  elsif !good_amount?(user_amount, resource)
```